src/rhsmlib/services/unregister.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import print_function, division, absolute_import
+
+#
+# Copyright (c) 2017 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+#
+
+"""
+This module includes class used for unregistering system from candlepin
+server.
+"""
+
+import logging
+
+from subscription_manager import injection as inj
+from subscription_manager import managerlib
+from rhsm import connection
+
+
+log = logging.getLogger(__name__)
+
+
+class UnRegister(object):
+    """
+    Class providing functionality of unregistering the system from
+    Candlepin server.
+    """
+
+    def __init__(self, uep, consumer_uuid):
+        """
+        Initialization of UnRegister instance
+        """
+        self.plugin_manager = inj.require(inj.PLUGIN_MANAGER)
+        self.uep = uep
+        self.consumer_uuid = consumer_uuid
+
+    def unregister(self):
+        """
+        Try to unregister the system from candlepin server
+        :return: None
+        """
+
+        self.plugin_manager.run(
+            "pre_unregister"
+        )
+
+        try:
+            self.uep.unregisterConsumer(self.consumer_uuid)
+            log.info("Successfully un-registered.")
+            managerlib.system_log("Unregistered machine with identity: %s" % self.consumer_uuid)
+            managerlib.clean_all_data(backup=False)
```

Comment:
We should also probably add a line to clear any connection information from cp_provider since the identity certificate is revoked after a consumer unregisters.